File: backend/app/models/delivery_zone.py
from datetime import datetime, timezone

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Integer, Numeric, String
from sqlalchemy.orm import relationship

from app.database import Base


class DeliveryZone(Base):
    __tablename__ = "delivery_zones"

    id = Column(Integer, primary_key=True, index=True)
    city_id = Column(Integer, ForeignKey("cities.id"), nullable=False, index=True)
    company_id = Column(Integer, ForeignKey("companies.id"), nullable=True, index=True)
    zone_name = Column(String, nullable=False)
    delivery_days = Column(String, nullable=False)
    extra_cost = Column(Numeric(10, 2), nullable=False, default=0)
    is_active = Column(Boolean, default=True, nullable=False)
    is_deleted = Column(Boolean, default=False, nullable=False)
    deleted_at = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), nullable=False)
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc), nullable=False)

    city = relationship("City", back_populates="delivery_zones")

File: frontend/types/deliveryZone.ts
export type DeliveryZone = {
  id: number
  city_id: number
  zone_name: string
  delivery_days: string
  extra_cost: number
  is_active: boolean
  is_deleted: boolean
  deleted_at: string | null
  created_at: string
  updated_at: string
}

export type DeliveryZoneCreatePayload = {
  city_id: number
  zone_name: string
  delivery_days: string
  extra_cost: number
  is_active: boolean
}

export type DeliveryZoneUpdatePayload = DeliveryZoneCreatePayload
